cms/djangoapps/contentstore/views/item.py

```diff
@@ -145,12 +146,25 @@ def xblock_handler(request, usage_key_string):
             accept_header = request.META.get('HTTP_ACCEPT', 'application/json')
 
             if 'application/json' in accept_header:
+                store = modulestore()
                 fields = request.GET.get('fields', '').split(',')
+                format = request.GET.get('format', '')
                 if 'graderType' in fields:
                     # right now can't combine output of this w/ output of _get_module_info, but worthy goal
                     return JsonResponse(CourseGradingModel.get_section_grader_type(usage_key))
+                elif format == 'concise':
+                    xblock = _get_xblock(usage_key, request.user)
+                    course = store.get_course(xblock.location.course_key)   # pylint: disable=no-member
+                    course_outline = create_xblock_info(
+                        course,
+                        include_child_info=True,
+                        include_children_predicate=lambda xblock: xblock.has_children,
+                        is_concise=format
+                    )
+                    ancestor_info = _create_xblock_ancestor_info(xblock, course_outline, is_concise=format)
```

Comment:
@mushtaqak I did a quick analysis of ``_create_xblock_ancestor_info`` and I don't see how it is using the course outline. It is passing it around to various methods but none of them seem to do much with it. You could try passing ``None`` and see what happens.